### PR TITLE
feat: support auto https #27

### PR DESCRIPTION
**Pull request by lxm**
implement  #27 

config add:

```
server:
  domain:
    - example1.com
    - example2.com
  port: 8222
  sslport: 443

tls:
  enabled: true
  auto: true
  cacheDir: /opt/
  certPath: "/etc/domain/cert.pem"
  certkeyPath: "/etc/domain/cert.key"
```

sslport only support 443 for now

### CODE REVIEW

**Comment by saltbo** [comment]
Please add the documentation about this feature.

File: config/config.go [hunk 1]
```diff
@@ -18,6 +18,22 @@ type Config struct {
 	Email      mailutil.Config
 	Database   gormutil.Config
 	Provider   provider.Config
+	Server     Server
+	TLS        TLS
```
Comment (saltbo) [inline config/config.go:22]: Could merge the Server and the TLS?

File: cmd/server.go [hunk 2]
```diff
@@ -84,5 +89,49 @@ func serverRun(conf *config.Config) {
...
+	if conf.TLS.Enabled {
+		tlsAddr := fmt.Sprintf(":%d", conf.Server.SSLPort)
+		go startTls(ge, tlsAddr, conf.TLS.Auto, conf.TLS.CacheDir, conf.Server.Domain, conf.TLS.CertPath, conf.TLS.CertkeyPath)
+	}
+
+	addr := fmt.Sprintf(":%d", conf.Server.Port)
+	ginutil.Startup(ge, addr)
+}
+
+func startTls(e *gin.Engine, addr string, auto bool, cacheDir string, domain []string, cert, key string) {
```
Comment (saltbo) [inline cmd/server.go:101]: It is best not to define so many parameters, you can directly pass in conf

File: config/config.go (hunk 1 above)
Comment (lxm) [reply config/config.go:22]: I don't think merging Server and TLS is a good idea

Server section can be used to set configs of the service, not only be used for tls

File: cmd/server.go (hunk 2 above)
Comment (lxm) [reply cmd/server.go]: changed the code

**Review by saltbo — APPROVED** [review]

**Pull request merged**